data.rs

```diff
@@ -267,4 +269,115 @@ impl PSP22Data {
             value,
         }])
     }
+
+    /// Burns `value` tokens from `from` account.
+    pub fn burn_from(&mut self,
```

Comment:
A lot of code in this function replicates `fn burn` - the only additional place here is the check and modification of allowances - so I'd change to code to reuse `fn burn` (i.e. call it and only add necessary allowances management around it).